tests/test_features/test_base.py

```diff
@@ -18,129 +18,444 @@
 ##########################################################################
 
 import pytest
+import numpy as np
+import numpy.testing as npt
 
+from ..fixtures import Dataset
 from yellowbrick.base import Visualizer
 from yellowbrick.features.base import *
-from tests.base import VisualTestCase
-from ..fixtures import Dataset
 
 from sklearn.base import BaseEstimator, TransformerMixin
 from sklearn.datasets import make_classification, make_regression
 
+from unittest.mock import Mock
+
+try:
+    import pandas as pd
+except ImportError:
+    pd = None
+
+
 ##########################################################################
 ## Fixtures
 ##########################################################################
 
 @pytest.fixture(scope='class')
 def discrete(request):
     """
-    Creare a random classification fixture.
+    Create a random classification dataset fixture.
     """
     X, y = make_classification(
-        n_samples=400, n_features=12, n_informative=10, n_redundant=0,
-        n_classes=5, random_state=2019)
+        n_classes=5,
+        n_samples=400,
+        n_features=12,
+        n_informative=10,
+        n_redundant=0,
+        random_state=2019
+    )
 
-    # Set a class attribute for digits
+    # Dataset is accessible on the class so it is only generated once
     request.cls.discrete = Dataset(X, y)
 
+
 @pytest.fixture(scope='class')
 def continuous(request):
     """
-    Creates a random regressor fixture.
+    Creates a random regression dataset fixture.
     """
     X, y = make_regression(
-        n_samples=500, n_features=22, n_informative=8, random_state=2019
+        n_samples=500,
+        n_features=22,
+        n_informative=8,
+        random_state=2019
     )
 
-    # Set a class attribute for digits
+    # Dataset is accessible on the class so it is only generated once
     request.cls.continuous = Dataset(X, y)
 
 
 ##########################################################################
 ## FeatureVisualizer Base Tests
 ##########################################################################
 
-class TestFeatureVisualizerBase(VisualTestCase):
+@pytest.mark.usefixtures("discrete")
+class TestFeatureVisualizer(object):
```

Comment:
I extended the tests for the `FeatureVisualizer` to make the whole subsection of the class hierarchy more robust.